kolibri/core/content/api.py

```diff
@@ -375,7 +386,31 @@ def filter_options(self, request, **kwargs):
 
 @method_decorator(remote_metadata_cache, name="dispatch")
 class ChannelMetadataViewSet(BaseChannelMetadataMixin, RemoteViewSet):
-    pass
+    # Overriding to isolate Public API from internal API
+    values = tuple(
+        field for field in BaseChannelMetadataMixin.values if field != "thumbnail"
+    )
+
+    def consolidate(self, items, queryset):
+        included_languages = {}
+        for (
+            channel_id,
+            language_id,
+        ) in models.ChannelMetadata.included_languages.through.objects.filter(
+            channelmetadata__in=queryset
+        ).values_list(
+            "channelmetadata_id", "language_id"
+        ):
+            if channel_id not in included_languages:
+                included_languages[channel_id] = []
+            included_languages[channel_id].append(language_id)
+        for item in items:
+            item["included_languages"] = included_languages.get(item["id"], [])
+            item["last_published"] = item["last_updated"]
+            item["thumbnail"] = reverse(
```

Comment:
I think this might be doable using the `field_map` class property which allows setting to a callable, that would allow the rest of the code not to be copy pasted here.